src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.cs

```diff
@@ -219,5 +224,23 @@ private static AbstractPersistentStorageService GetPersistentStorageService()
             var persistentStorageService = workspace.Services.GetService<IPersistentStorageService>() as AbstractPersistentStorageService;
             return persistentStorageService;
         }
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+        private static extern IntPtr AddDllDirectory(string directory);
+
+        private static void SetNativeDllSearchDirectories()
+        {
+            // Set LoadLibrary search directory to %VSINSTALLDIR%\Common7\IDE so that the compiler 
+            // can P/Invoke to Microsoft.DiaSymReader.Native when emitting Windows PDBs.
+
+            Contract.Assert(string.Equals(Path.GetFileName(
+                AppDomain.CurrentDomain.BaseDirectory.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar)), "IDE", StringComparison.OrdinalIgnoreCase));
+
+            var cookie = AddDllDirectory(AppDomain.CurrentDomain.BaseDirectory);
+            if (cookie == IntPtr.Zero)
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
```

Comment:
Why risk the data loss of converting a Win32 error to an HRESULT before throwing? Why now just `throw new Win32Exception();` ? That will automatically look up the last win32 error and throw with the best message. 
Or perhaps your goal is that `ThrowExceptionForHR` may throw a more specific exception type? That's possible I suppose.